src/main/kotlin/org/cascadebot/bot/rabbitmq/actions/AutoResponderProcessor.kt

```diff
@@ -56,13 +55,30 @@ class AutoResponderProcessor : Processor {
         }
 
         return RabbitMQResponse.success(
-            AutoResponderResponse(
-                slot.slotId,
-                slot.slotType,
-                autoResponder.enabled,
-                autoResponder.text,
-                autoResponder.match
-            )
+            AutoResponderResponse.fromEntity(autoResponder)
         )
     }
+
+    private fun updateAutoResponder(
+        body: ObjectNode,
+        guild: Guild
+    ): RabbitMQResponse<AutoResponderResponse> {
+        val updateRequest = Main.json.treeToValue<UpdateAutoResponderRequest>(body)
+
+        val autoResponder = dbTransaction {
+            get(AutoResponderEntity::class.java, updateRequest.slotId)
+        }
+
+        if (autoResponder == null || autoResponder.slot.guildId != guild.idLong) {
+            return CommonResponses.AUTORESPONDER_NOT_FOUND
+        }
+
+        autoResponder.apply {
+            text = updateRequest.text
+            match = updateRequest.matchText.toMutableList()
+            enabled = updateRequest.enabled
```

Comment:
While it would definitely be nice, I dont see it as a necessity. I can work with full data replacements instead of partials.

Reply:
I do agree partial updating would be nice, but it would add a lot of complexity on the bot side for handling this data - hence why I went with full updating

Reply:
Fair enough, we should add a note to the notion, or a github issue, or something reminding us to revisit this in the future, but I'm good with full updates for now.